sound/soc/intel/boards/Kconfig

```diff
@@ -485,6 +485,7 @@ config SND_SOC_INTEL_SOF_RT5682_MACH
 	select SND_SOC_RT1011
 	select SND_SOC_RT1015
 	select SND_SOC_RT1015P
+	select SND_SOC_RT1019
```

Comment:
THIS CANNOT WORK.

in rt1015p.c, I see this
````c
#ifdef CONFIG_ACPI
static const struct acpi_device_id rt1015p_acpi_match[] = {
	{ "RTL1015", 0},
	{ "RTL1019", 0},
	{ },
};
MODULE_DEVICE_TABLE(acpi, rt1015p_acpi_match);
#endif
````

in rt1019.c, I see this
````c
#ifdef CONFIG_ACPI
static const struct acpi_device_id rt1019_acpi_match[] = {
	{ "10EC1019", 0},
	{ },
};
MODULE_DEVICE_TABLE(acpi, rt1019_acpi_match);
#endif
````

You cannot have a build where you have two codec drivers probing for the same ACPI ID. That's just wrong. You need to fix this!

Reply:
Looks like rt1019.c is i2c mode driver and rt1015p.c is auto mode driver for this ALC1019 amplifier.

obj-$(CONFIG_SND_SOC_RT1015P)	+= snd-soc-rt1015p.o
obj-$(CONFIG_SND_SOC_RT1019)	+= snd-soc-rt1019.o

It seems SND_SOC_RT1015P is sufficient to build the auto mode driver, we don't need SND_SOC_RT1019 here.

There is restrictions about the I2S setting for amplifier running in auto mode. We may need to check the sof-adl-rt1019-rt5682.tplg against ALC1019's datasheet.

Reply:
BTW, I used bclk 3.072MHz, I2S, 32-bit slot-width, 24-bit data-width for ALC1015 running in auto-mode. Don't know if it's still valid for ALC1019 auto-mode.

Reply:
Yes we want to use auto/easy mode. My bad i dint check  SND_SOC_RT1019  is used to compile rt1019.c 
let me remove SND_SOC_RT1019 from Kconfig.

Reply:
> There is restrictions about the I2S setting for amplifier running in auto mode. We may need to check the sof-adl-rt1019-rt5682.tplg against ALC1019's datasheet.

we are using 1.536MHz in topology which is supported by ALC1019 (mentioned in datasheet)

Reply:
@Vamshigopal can we align on the configuration used by @brentlu ? Fewer configurations are better!

Reply:
@plbossart We are aligned on the same configuration used by @brentlu.  PR #5748